### PR TITLE
Opt-Space no longer scrolls (Chrome OSX)

### LINKED ISSUE #19: Pressing Opt-Space scrolls the page down
If you type Opt-Space to insert a nbsp (on http://facebook.github.io/draft-js/, for example) in Chrome Mac, the page scrolls down a screenful but it probably shouldn't move. Missing preventDefault() somewhere probably.

### PR DESCRIPTION
**Pull request by brookslyrette**
`Option + Space` no longer causes scrolling. This was only happening in Chrome/OSX. This was happening in the keyDown handler not the keyPress handler. 

I'm not sure if I like adding the ' ' to the editor in editOnKeyDown. Maybe I should implement this as a command?

Fixes #19

### CODE REVIEW

File: src/component/handlers/edit/editOnKeyDown.js [hunk 1]
```diff
@@ -101,6 +107,19 @@ function editOnKeyDown(e: SyntheticKeyboardEvent): void {
...
+      // handling for OSX where option + space scrolls
+      if (isChrome && isOptionKeyCommand(e)) {
+        e.preventDefault();
+        // insert a space into the editor
+        const contentState = DraftModifier.replaceText(
+            editorState.getCurrentContent(),
+            editorState.getSelection(),
+            ' '
+        );
+        this.update(EditorState.push(editorState, contentState, 'space'));
```
Comment (hellendag) [inline src/component/handlers/edit/editOnKeyDown.js]: The correct `EditorChangeType` here is `'insert-characters'`. (I need to document this, but one should always stick to the `EditorChangeType` values for `push` calls.)

File: src/component/handlers/edit/editOnKeyDown.js [hunk 2]
```diff
@@ -101,6 +107,19 @@ function editOnKeyDown(e: SyntheticKeyboardEvent): void {
...
       return;
+    case Keys.SPACE:
+      // handling for OSX where option + space scrolls
+      if (isChrome && isOptionKeyCommand(e)) {
+        e.preventDefault();
+        // insert a space into the editor
+        const contentState = DraftModifier.replaceText(
+            editorState.getCurrentContent(),
+            editorState.getSelection(),
+            ' '
```
Comment (hellendag) [inline src/component/handlers/edit/editOnKeyDown.js]: Spacing nit: FB uses two spaces for indents.

**Comment by hellendag** [comment]
@brookslyrette on the import I ran for your last PR, our bot pulled in changes that had already been merged, I assume because the commits were still present in this stack. Can those older changes be rebased out of this PR?

Regarding the PR itself, I'm wondering if this should be handled within `getDefaultKeyBinding`. It would be functionally the same as handling it in `editOnKeyDown`, but at least it would be with other key bindings that check OS/browser combinations.

**Comment by brookslyrette** [comment]
Thanks @hellendag. I'll make the corrections and rebase the PR. 

I had attempted to prevent the scrolling in `getDefaultKeyBinding` but since I was editing the editorState I placed it in `editOnKeyDown`. From what I can tell, I can't add the space to the editor state from `getDefaultKeyBinding` without creating a command. 

I'm happy to do so if you think it's a cleaner approach. What name would be appropriate for the command?

**Comment by hellendag** [comment]
Oh of course, that's a great point. There would have to be a command string for it. Let's stick with what you have.

Were you able to determine what's actually causing this? I.e. why would Draft behave differently from a plain contenteditable?

**Comment by brookslyrette** [comment]
As far as I can tell it is because `document.activeElement` is not the contentEditable area. It's that the div is focused: 

screen shot 2016-03-29 at 12 50 20 pm

That seemed a bit too fundamental for me to be messing around with.

**Comment by hellendag** [comment]
Hmm, though that div is the `contenteditable`.

> That seemed a bit too fundamental for me to be messing around with.

No worries, I'll see if I can track it down. :)

**Comment by brookslyrette** [comment]
Just a hunch. This only seems to happen when `Option + Space` is entered into an empty editor. 

Not sure if it's a redraw or the placeholder causing it but I'm sure it has to do with the initial state.

**Comment by hellendag** [comment]
It happens whenever we have to `preventDefault` on the `textInput` event, i.e. when we need to render the character manually rather than allowing the native behavior. This will happen at the start of a paragraph, after an immutable entity, and other cases.

It seems that if the `textInput` event is prevented when the `keydown` is allowed, the scroll behavior occurs.

Since we need to prevent the `textInput` event in the case described above, I think we don't really have any choice but to prevent the `keydown` and perform the manual insertion as you've done here.

**Comment by hellendag** [comment]
Repro: https://jsfiddle.net/hellendag/amLxv1w8/

File: src/component/handlers/edit/editOnKeyDown.js [hunk 3]
```diff
@@ -101,6 +107,19 @@ function editOnKeyDown(e: SyntheticKeyboardEvent): void {
...
       return;
+    case Keys.SPACE:
+      // handling for OSX where option + space scrolls
+      if (isChrome && isOptionKeyCommand(e)) {
+        e.preventDefault();
+        // insert a space into the editor
+        const contentState = DraftModifier.replaceText(
+          editorState.getCurrentContent(),
+          editorState.getSelection(),
+          ' '
```
Comment (hellendag) [inline src/component/handlers/edit/editOnKeyDown.js]: Is this the correct character to insert here, if the goal of <kbd>Opt</kbd>+<kbd>Space</kbd> is to insert a non-breaking space? Or should this be `'\u00a0'`?

**Comment by hellendag** [comment]
Thanks!

**Comment by hellendag** [comment]
@facebook-github-bot import

**Pull request closed without merging**